design-proposals/vm-pool.md

```diff
@@ -0,0 +1,194 @@
+# Overview
+
+This design provides an approach for creating a VM grouping and replication abstraction for KubeVirt called a VirtualMachinePool. 
+
+## Motivation
+
+The ability to manage groups (or fleets) of similar VMs using a higher level abstraction is a staple among commonly utilized Iaas operational patterns. By bringing stateful VM group management to KubeVirt, we open the door for operation teams to utilize their existing patterns for managing KubeVirt VMs. This feature further aligns KubeVirt as an Iaas offering comparable to the public clouds which makes KubeVirt a more attractive option for Iaas management on baremetal hardware.
+
+## Goals
+
+* VM pool abstraction capable of managing replication of stateful VMs at scale
+* Automated rollout of spec changes across and other updates across a pool of stateful VMs
+* Automated and manual scale-out and scale-in VMs associated with a pool
+* Autohealing (delete and replace) of VMs in a pool not passing health checks
+* Ability to specify unique secrets and configMap data per VM within a VMPool
+
+## Non-Goals
+
+* Not designing a pool abstraction capable of managing multiple VM shapes within a single pool. A VMPool consists only of VMs replicated from a single config.
+
+## Terms
+
+* **VirtualMachine [VM]** - Refers specifically to the KubeVirt VirtualMachine api object. 
+* **VirtualMachinePool [VMPool]** - KubeVirt API for scaling out/in replicas of KubeVirt VirtualMachines
+* **VirtualMachineConfig [VMConfig]** - KubeVirt API representing a VirtualMachine spec which is referenced by a VirtualMachinePool.
+* **Scale-out and Scale-in** - Terms to describe the action of adding/removing to the count of VMs within a VMPool.
```

Comment:
Are Scale-in and Scale-out common terms for that kind of pattern?
At first reading I mixed them up as I read scale-in as "scaling VMs into the pool -> growing" which is obviously wrong.
What about Scale-Up and down, I think that's also very known terminology.

Reply:
scale-out and scale-up  have been used interchangeably in the past. I see that scale-out is becoming more consistent in conversations where we're scaling horizontally.  scale-up can also refer to vertical scaling, which is why i'm beginning to prefer scale-out